IsDockerWorking not using TestContainersStrategy after 2.13.4.Final

### Describe the bug

There is a regression in the class `IsDockerWorking`, it was detected running on GitLab CI using the docker socket configuration: https://www.testcontainers.org/supported_docker_environment/continuous_integration/gitlab_ci/

The first strategy should be using the `TestContainersStrategy` and is able to detect the socket correctly.

I tested it with and without a `~/.testcontainers.properties`, but it's the same behavior.

The only related change is https://github.com/quarkusio/quarkus/pull/28702

cc: @geoand 

### Expected behavior

Running Quarkus 2.13.3.Final:

```
2022-11-10 18:36:35,116 DEBUG [io.qua.dep.IsDockerWorking] (build-40) Checking Docker Environment using strategy io.quarkus.deployment.IsDockerWorking$TestContainersStrategy
2022-11-10 18:36:35,161 INFO  [org.tes.doc.DockerClientProviderStrategy] (build-40) Loaded org.testcontainers.dockerclient.UnixSocketClientProviderStrategy from ~/.testcontainers.properties, will try it first
2022-11-10 18:36:36,001 INFO  [org.tes.doc.DockerClientProviderStrategy] (build-40) Found Docker environment with local Unix socket (unix:///var/run/docker.sock)
```
Only the TestContainersStrategy runs and works correctly.

### Actual behavior

Running Quarkus 2.13.3.Final and 2.14.0.Final:

On the docker.sock mounted CI pipeline failing:
```
2022-11-10 17:19:04,262 DEBUG [io.qua.dep.IsDockerWorking] (build-8) Checking Docker Environment using strategy io.quarkus.deployment.IsDockerWorking$TestContainersStrategy
2022-11-10 17:19:04,285 DEBUG [io.qua.dep.IsDockerWorking] (build-8) Checking Docker Environment using strategy io.quarkus.deployment.IsDockerWorking$DockerHostStrategy
2022-11-10 17:19:04,286 DEBUG [io.qua.dep.IsDockerWorking] (build-8) Checking Docker Environment using strategy io.quarkus.deployment.IsDockerWorking$DockerBinaryStrategy
2022-11-10 17:19:04,321 WARN  [io.qua.dep.IsDockerWorking] (build-8) No docker binary found or general error: java.lang.RuntimeException: Input/Output error while executing command.
```

On my local laptop fallback to using the docker binary detection:
```
2022-11-10 18:21:17,864 DEBUG [io.qua.dep.IsDockerWorking] (build-36) Checking Docker Environment using strategy io.quarkus.deployment.IsDockerWorking$TestContainersStrategy
2022-11-10 18:21:17,886 DEBUG [io.qua.dep.IsDockerWorking] (build-36) Checking Docker Environment using strategy io.quarkus.deployment.IsDockerWorking$DockerHostStrategy
2022-11-10 18:21:17,887 DEBUG [io.qua.dep.IsDockerWorking] (build-36) Checking Docker Environment using strategy io.quarkus.deployment.IsDockerWorking$DockerBinaryStrategy
2022-11-10 18:21:17,969 DEBUG [io.qua.dep.IsDockerWorking] (build-36) Docker daemon found. Version: '20.10.21'
2022-11-10 18:21:17,987 INFO  [org.tes.uti.ImageNameSubstitutor] (build-36) Image name substitution will be performed by: DefaultImageNameSubstitutor (composite of 'ConfigurationFileImageNameSubstitutor' and 'PrefixingImageNameSubstitutor')
2022-11-10 18:21:18,007 INFO  [org.tes.doc.DockerClientProviderStrategy] (build-36) Loaded org.testcontainers.dockerclient.UnixSocketClientProviderStrategy from ~/.testcontainers.properties, will try it first
2022-11-10 18:21:18,816 INFO  [org.tes.doc.DockerClientProviderStrategy] (build-36) Found Docker environment with local Unix socket (unix:///var/run/docker.sock)
```

It runs the strategies TestContainersStrategy, DockerHostStrategy, DockerBinaryStrategy... here it works since it find the docker binary, yet TestContainers is using the UnixSocketClientProviderStrategy.

### How to Reproduce?

_No response_

### Output of `uname -a` or `ver`

_No response_

### Output of `java -version`

openjdk 17.0.5 2022-10-18

### GraalVM version (if different from Java)

_No response_

### Quarkus version or git rev

2.13.4.Final

### Build tool (ie. output of `mvnw --version` or `gradlew --version`)

Apache Maven 3.8.6

### Additional information

_No response_

With upgrading the quarkus version from 2.13.3.Final to 2.13.4.Final and 2.14.0.Final, same error is happening.

Even if devservices datasource configurations exist, we get following error:

`java.lang.RuntimeException: java.lang.RuntimeException: io.quarkus.builder.BuildException: Build failure: Build failed due to errors
	[error]: Build step io.quarkus.datasource.deployment.devservices.DevServicesDatasourceProcessor#launchDatabases threw an exception: java.lang.IllegalStateException: Please configure the datasource URL for  datasource 'test-db' or ensure the Docker daemon is up and running.
	at io.quarkus.datasource.deployment.devservices.DevServicesDatasourceProcessor.startDevDb(DevServicesDatasourceProcessor.java:247)
	at io.quarkus.datasource.deployment.devservices.DevServicesDatasourceProcessor.launchDatabases(DevServicesDatasourceProcessor.java:142)
	at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
	at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62)
	at java.base/jdk.internal.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
	at java.base/java.lang.reflect.Method.invoke(Method.java:566)
	at io.quarkus.deployment.ExtensionLoader$3.execute(ExtensionLoader.java:909)
	at io.quarkus.builder.BuildContext.run(BuildContext.java:281)
	at org.jboss.threads.ContextHandler$1.runWith(ContextHandler.java:18)
	at org.jboss.threads.EnhancedQueueExecutor$Task.run(EnhancedQueueExecutor.java:2449)
	at org.jboss.threads.EnhancedQueueExecutor$ThreadBody.run(EnhancedQueueExecutor.java:1478)
	at java.base/java.lang.Thread.run(Thread.java:834)
	at org.jboss.threads.JBossThread.run(JBossThread.java:501)

	at io.quarkus.test.junit.QuarkusTestExtension.throwBootFailureException(QuarkusTestExtension.java:625)
	at io.quarkus.test.junit.QuarkusTestExtension.interceptTestClassConstructor(QuarkusTestExtension.java:696)`
	
I tried with enabling devservices on test profile and trigger the build without running tests, following error is happening:

`[WARNING] [io.quarkus.deployment.IsDockerWorking] 'docker -v' returned an error code. Make sure your Docker binary is correct
[INFO] ------------------------------------------------------------------------
[INFO] BUILD FAILURE
[INFO] ------------------------------------------------------------------------
[INFO] Total time:  18.425 s
[INFO] Finished at: 2022-11-11T11:26:08Z
[INFO] ------------------------------------------------------------------------
[ERROR] Failed to execute goal io.quarkus:quarkus-maven-plugin:2.14.0.Final:build (default) on project ***: Failed to build quarkus application: io.quarkus.builder.BuildException: Build failure: Build failed due to errors
[ERROR] 	[error]: Build step io.quarkus.container.image.docker.deployment.DockerProcessor#dockerBuildFromJar threw an exception: java.lang.RuntimeException: Unable to build docker image. Please check your docker installation
[ERROR] 	at io.quarkus.container.image.docker.deployment.DockerProcessor.dockerBuildFromJar(DockerProcessor.java:87)
[ERROR] 	at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
[ERROR] 	at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:77)
[ERROR] 	at java.base/jdk.internal.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
[ERROR] 	at java.base/java.lang.reflect.Method.invoke(Method.java:568)
[ERROR] 	at io.quarkus.deployment.ExtensionLoader$3.execute(ExtensionLoader.java:909)
[ERROR] 	at io.quarkus.builder.BuildContext.run(BuildContext.java:281)
[ERROR] 	at org.jboss.threads.ContextHandler$1.runWith(ContextHandler.java:18)
[ERROR] 	at org.jboss.threads.EnhancedQueueExecutor$Task.run(EnhancedQueueExecutor.java:2449)
[ERROR] 	at org.jboss.threads.EnhancedQueueExecutor$ThreadBody.run(EnhancedQueueExecutor.java:1478)
[ERROR] 	at java.base/java.lang.Thread.run(Thread.java:833)
[ERROR] 	at org.jboss.threads.JBossThread.run(JBossThread.java:501)`

However, running the build second time before spot fleet instance is closed, it gives successful build and image is pushed.

With upgrading the quarkus version from 2.13.3.Final to 2.13.4.Final and 2.14.0.Final, same error is happening.

Even if devservices datasource configurations exist, we get following error:

```
java.lang.RuntimeException: java.lang.RuntimeException: io.quarkus.builder.BuildException: Build failure: Build failed due to errors
[error]: Build step io.quarkus.datasource.deployment.devservices.DevServicesDatasourceProcessor#launchDatabases threw an exception: java.lang.IllegalStateException: Please configure the datasource URL for datasource 'test-db' or ensure the Docker daemon is up and running.
at io.quarkus.datasource.deployment.devservices.DevServicesDatasourceProcessor.startDevDb(DevServicesDatasourceProcessor.java:247)
at io.quarkus.datasource.deployment.devservices.DevServicesDatasourceProcessor.launchDatabases(DevServicesDatasourceProcessor.java:142)
at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62)
at java.base/jdk.internal.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
at java.base/java.lang.reflect.Method.invoke(Method.java:566)
at io.quarkus.deployment.ExtensionLoader$3.execute(ExtensionLoader.java:909)
at io.quarkus.builder.BuildContext.run(BuildContext.java:281)
at org.jboss.threads.ContextHandler$1.runWith(ContextHandler.java:18)
at org.jboss.threads.EnhancedQueueExecutor$Task.run(EnhancedQueueExecutor.java:2449)
at org.jboss.threads.EnhancedQueueExecutor$ThreadBody.run(EnhancedQueueExecutor.java:1478)
at java.base/java.lang.Thread.run(Thread.java:834)
at org.jboss.threads.JBossThread.run(JBossThread.java:501)

at io.quarkus.test.junit.QuarkusTestExtension.throwBootFailureException(QuarkusTestExtension.java:625)
at io.quarkus.test.junit.QuarkusTestExtension.interceptTestClassConstructor(QuarkusTestExtension.java:696)
```

I tried with enabling devservices on test profile and trigger the build without running tests, following error is happening:

```
[WARNING] [io.quarkus.deployment.IsDockerWorking] 'docker -v' returned an error code. Make sure your Docker binary is correct
[INFO] ------------------------------------------------------------------------
[INFO] BUILD FAILURE
[INFO] ------------------------------------------------------------------------
[INFO] Total time:  18.425 s
[INFO] Finished at: 2022-11-11T11:26:08Z
[INFO] ------------------------------------------------------------------------
[ERROR] Failed to execute goal io.quarkus:quarkus-maven-plugin:2.14.0.Final:build (default) on project ***: Failed to build quarkus application: io.quarkus.builder.BuildException: Build failure: Build failed due to errors
[ERROR] 	[error]: Build step io.quarkus.container.image.docker.deployment.DockerProcessor#dockerBuildFromJar threw an exception: java.lang.RuntimeException: Unable to build docker image. Please check your docker installation
[ERROR] 	at io.quarkus.container.image.docker.deployment.DockerProcessor.dockerBuildFromJar(DockerProcessor.java:87)
[ERROR] 	at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
[ERROR] 	at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:77)
[ERROR] 	at java.base/jdk.internal.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
[ERROR] 	at java.base/java.lang.reflect.Method.invoke(Method.java:568)
[ERROR] 	at io.quarkus.deployment.ExtensionLoader$3.execute(ExtensionLoader.java:909)
[ERROR] 	at io.quarkus.builder.BuildContext.run(BuildContext.java:281)
[ERROR] 	at org.jboss.threads.ContextHandler$1.runWith(ContextHandler.java:18)
[ERROR] 	at org.jboss.threads.EnhancedQueueExecutor$Task.run(EnhancedQueueExecutor.java:2449)
[ERROR] 	at org.jboss.threads.EnhancedQueueExecutor$ThreadBody.run(EnhancedQueueExecutor.java:1478)
[ERROR] 	at java.base/java.lang.Thread.run(Thread.java:833)
[ERROR] 	at org.jboss.threads.JBossThread.run(JBossThread.java:501)
```

However, running the build second time before spot fleet instance is closed, it gives successful build and image is pushed.

--- a/core/deployment/src/main/java/io/quarkus/deployment/IsDockerWorking.java
+++ b/core/deployment/src/main/java/io/quarkus/deployment/IsDockerWorking.java
@@ -84,7 +84,7 @@ public class IsDockerWorking implements BooleanSupplier {
 
                 Class<?> configurationClass = Thread.currentThread().getContextClassLoader()
                         .loadClass("org.testcontainers.utility.TestcontainersConfiguration");
-                Object configurationInstance = configurationClass.getMethod("instance").invoke(null);
+                Object configurationInstance = configurationClass.getMethod("getInstance").invoke(null);
                 String oldReusePropertyValue = (String) configurationClass
                         .getMethod("getUserProperty", String.class, String.class)
                         .invoke(configurationInstance, "testcontainers.reuse.enable", "false"); // use the default provided in TestcontainersConfiguration#environmentSupportsReuse

